src/snapred/backend/recipe/algorithm/SmoothDataExcludingPeaksAlgo.py

```diff
@@ -0,0 +1,111 @@
+# Logic notes:
+"""
+    load nexus
+    num spe = num groups (instrument state) --> consitant with workspace
+    state & crystal info --> consitant with workspace
+    call diffraction spectrum peak predictor --> to get peaks (num groups)
+    call diffraction spectrum weight calc (send one spectra only)
+    extract x and y data from a workspace (numpy arrray)
+    implement csaps
+    create new workspace with csaps data
+"""
+import json
+
+from mantid.api import (
+    AlgorithmFactory,
+    PythonAlgorithm,
+    WorkspaceGroup,
+    mtd,
+)
+from mantid.kernel import Direction
+from scipy.interpolate import splev, splrep
+
+from snapred.backend.dao.SmoothDataExcludingPeaksIngredients import SmoothDataExcludingPeaksIngredients
+from snapred.backend.recipe.algorithm.DiffractionSpectrumWeightCalculator import DiffractionSpectrumWeightCalculator
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+name = "SmoothDataExcludingPeaks"
+
+
+class SmoothDataExcludingPeaks(PythonAlgorithm):
+    def PyInit(self):
+        # declare properties
+        self.declareProperty("InputWorkspace", defaultValue="", direction=Direction.Input)
+        self.declareProperty("SmoothDataExcludingPeaksIngredients", defaultValue="", direction=Direction.Input)
+        self.declareProperty("OutputWorkspace", defaultValue="", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def PyExec(self):
+        self.log().notice("Removing peaks and smoothing data")
+
+        # load ingredients
+        smoothdataIngredients = SmoothDataExcludingPeaksIngredients(
+            **json.loads(self.getProperty("SmoothDataExcludingPeaksIngredients").value)
+        )
+
+        # load workspace
+        input_ws = self.getProperty("InputWorkspace").value
+        self.mantidSnapper.mtd[input_ws]
```

Comment:
This line is asking the ADS to look for the workspace with the string name in `input_ws`, but then doesn't assign the returned pointer.  So I think it can be deleted.

Reply:
If I delete this then how can the "InputWorkspace" property be sent to other calls as a variable if it is not set to any variables? Wouldn't I have to use `self.getProperty("InputWorkspace).value` for every place where `input_ws` is being used?

Reply:
The line you can delete is L49, `self.mantifSnapper.mtd[input_ws]`, since it doesn't do anything.  L48 is good as it is.

Reply:
Gotcha!